### PR TITLE
Add website authentication

### LINKED ISSUE #536: Auth -> Sessions example should use Basic auth
The current auth sessions docs show setting up sessions + a bearer token: https://docs.vapor.codes/4.0/authentication/#session

This doesn't make a whole lot of sense since usually sessions are paired with a single login route. Showing basic auth, credential auth, or manual auth would make more sense here. 

This requires two routes groups looking like:

```swift
// Routes that use sessions
let sessions = admin.grouped([
    app.sessions.middleware,
    User.sessionAuthenticator(),
])
sessions.post("/login") { req in 
    // Some stuff
    req.auth.login(theUser)
}

// Routes that require login
let protected = sessions.grouped([
    User.redirectMiddleware(path: "/admin/login")
])
protected.get("/me") { req in 
    try req.auth.require(User.self)
}
```

### PR DESCRIPTION
**Pull request by 0xTim**
Add an explanation for how to do authentication with a website.

Resolves #536 

Dependent on vapor/fluent#711 being merged first

### CODE REVIEW

File: 4.0/docs/authentication.md [hunk 1]
```diff
@@ -673,6 +673,77 @@ This will use the application's default database for resolving the user. To spec
...
+```
+
+This uses the default credentials authenticator to protect the login route. You must send `username` and `password` in the POST request. You can set your form up like so:
+
+```html
+ <form method="POST" action="/login">
+    <label for="username">Username</label>
+    <input type="text" id="username" placeholder="Username" name="username" autocomplete="username" required autofocus>
+    <label for="password">Password</label>
+    <input type="password" id="password" placeholder="Password" name="password" autocomplete="current-password" required>
```
Comment (jdmcd) [inline 4.0/docs/authentication.md:740]: Shouldn't this `autocomplete` be `password`?

**Review by jdmcd — CHANGES_REQUESTED** [review]
Just one nit

File: 4.0/docs/authentication.md (hunk 1 above)
Comment (0xTim) [reply 4.0/docs/authentication.md:740]: No, it's used by browsers to work out what is a current password and when to suggest a new one
https://developer.apple.com/documentation/security/password_autofill/enabling_password_autofill_on_an_html_input_element

Comment (jdmcd) [reply 4.0/docs/authentication.md:740]: Ah, ok got it. LGTM then!

Comment (0xTim) [reply 4.0/docs/authentication.md:740]: Want to click the approve button?

**Review by jdmcd — APPROVED** [review]

**Pull request merged**